### PR TITLE
ui-components: upgrade react/react-dom to v18

### PR DESCRIPTION
**Pull request by lassenordahl**
This PR just upgrades the React version used in ui-components to react 18, 
as well as the testing library version. Shouldn't have an impact on DB Console,
 since DB-Console is already using a React version newer than `ui-components`. No 
components in this Repo use new API's introduced in React 18.

Release note: None

### CODE REVIEW

File: packages/ui-components/package.json [hunk 1]
```diff
@@ -67,8 +70,8 @@
     "jest-environment-jsdom": "29.2.0",
     "jest-runner-eslint": "1.1.0",
     "prettier": "2.7.1",
-    "react": "17.0.2",
-    "react-dom": "17.0.2",
+    "react": "^18.2.0",
```
Comment (laurenbarker) [inline packages/ui-components/package.json:73]: @lassenordahl it looks like there are some dependencies listed twice.

**Review by laurenbarker — APPROVED** [review]
Once the dep list is cleaned up, this LGTM!

File: packages/ui-components/package.json (hunk 1 above)
Comment (lassenordahl) [reply packages/ui-components/package.json:73]: I'll remove it from `devDependencies`

-- https://stackoverflow.com/questions/18875674/whats-the-difference-between-dependencies-devdependencies-and-peerdependencies

**Comment by lassenordahl** [comment]
TFTR!

File: packages/ui-components/package.json [hunk 2]
```diff
@@ -29,7 +29,10 @@
   "dependencies": {
     "@cockroachlabs/icons": "^0.5.4",
     "@popperjs/core": "^2.9.2",
+    "@testing-library/react": "^13.4.0",
```
Comment (laurenbarker) [inline packages/ui-components/package.json:32]: This should only be in dev deps since it's used for testing.

**Pull request closed without merging**